src/app/service/data-service.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class DataService {
+  public data = [{
+    first: [
+      {
+        list: 'first',
+        id: 1,
+        title: 'First list title item',
+        description: 'Example description for first list title item',
+        completed: false,
+      },
+    ],
+    second: [
+      {
+        list: 'second',
+        id: 2,
+        title: 'Second list title item',
+        description: 'Example description for second list title item',
+        completed: false,
+      },
+    ],
+    third: [
+      {
+        list: 'third',
+        id: 3,
+        title: 'Third list title item',
+        description: 'Example description for third list title item',
+        completed: false,
+      },
+    ],
+  }];
+
+  loadAll() {
+    return this.data;
+  }
+  // Add an item in a specific list
+  addItem(list, item) {
+    if (list in this.data[0]) {
+      this.data[0][list].push(item);
+    } else {
+      this.addList(list, item);
+    }
+  }
+  // Add a list whit an item inside, or an empty list
+  addList(list, item) {
+    if (item) {
+      this.data[0][list] = [item];
+    } else {
+      this.data[0][list] = [];
+    }
+  }
+  // Remove an item form a specific list
+  removeItem(list, id) {
+    const index = this.data[0][list].map(x => {
+      return x.Id;
+    }).indexOf(id);
+    this.data[0][list].splice(index, 1);
```

Comment:
you can do all this code with a more elegant `filter`

`this.data[0][list] = this.data[0][list].filter((x) => x.Id !== id);`